Fix incorrect fix to a prior bug, which broke Connection.escape() method.
Per bug #331021.  See also upstream report at
http://sourceforge.net/tracker/index.php?func=detail&aid=1774101&group_id=22307&atid=374932


diff -Naur MySQL-python-1.2.2.orig/MySQLdb/connections.py MySQL-python-1.2.2/MySQLdb/connections.py
--- MySQL-python-1.2.2.orig/MySQLdb/connections.py	2007-02-24 19:10:53.000000000 -0500
+++ MySQL-python-1.2.2/MySQLdb/connections.py	2008-06-19 22:21:16.000000000 -0400
@@ -140,11 +140,10 @@
 
         conv2 = {}
         for k, v in conv.items():
-            if isinstance(k, int):
-                if isinstance(v, list):
-                    conv2[k] = v[:]
-                else:
-                    conv2[k] = v
+            if isinstance(k, int) and isinstance(v, list):
+                conv2[k] = v[:]
+            else:
+                conv2[k] = v
         kwargs2['conv'] = conv2
 
         self.cursorclass = kwargs2.pop('cursorclass', self.default_cursor)
